pkgs/by-name/si/simulide/package.nix

```diff
@@ -0,0 +1,79 @@
+{ lib
+, stdenv
+, fetchbzr
+, libsForQt5
+}:
+
+let
+  version = "0.4.15";
+  release = "SR10";
+  branch = "simulide_0.4.14"; # the branch name does not mach the version for some reason
+  rev = "291";
+  sha256 = "sha256-BBoZr/S2pif0Jft5wrem8y00dXl08jq3kFiIUtOr3LM=";
+in
+stdenv.mkDerivation {
+  pname = "simulide";
+  version = "${version}-${release}";
+
+  src = fetchbzr {
+    url = "https://code.launchpad.net/~arcachofo/simulide/${branch}";
+    inherit rev sha256;
+  };
+
+  postPatch = ''
+    sed -i resources/simulide.desktop \
+      -e "s|^Exec=.*$|Exec=simulide|" \
+      -e "s|^Icon=.*$|Icon=simulide|"
+    sed -i SimulIDE.pro \
+      -e "s|^VERSION = .*$|VERSION = ${version}|" \
+      -e "s|^RELEASE = .*$|RELEASE = -${release}|" \
+      -e "s|^REV_NO = .*$|REV_NO = ${rev}|" \
+      -e "s|^BUILD_DATE = .*$|BUILD_DATE = ??-??-??|"
+  '';
+
+  preConfigure = ''
+    cd build_XX
+  '';
+
+  nativeBuildInputs = with libsForQt5; [
```

Comment:
I think it's recommended to use `libsForQt5.callPackage` for qt5 packages, since it has a custom version of `mkDerivation` for use.

Reply:
I was wondering whether QT based apps should try to use `pkgs/by-name` or stay with the usual `qt*.callPackage`.

I personally would go with the `pkgs/by-name` one, but I can see the benefit of having the qt deps also acting like any other package being imported.

IIRC the QT `mkDerivation` has `wrapQtAppsHook` `qtbase` automatically included, but maybe there's some other useful things about it that I don't remember

Reply:
> I was wondering whether QT based apps should try to use `pkgs/by-name` or stay with the usual `qt*.callPackage`.

These two can co-exist, the package.nix stays in by-name, while an entry is added to all-packages.nix.

Reply:
IIRC I tried that once but it gave me some CI check fails.
Maybe I just misremember.
I can't currently re-check all-packages, but I don't recall seeing direct imports from pkgs/by-name in there.

Reply:
```
❯ rg by-name pkgs/top-level/all-packages.nix
824:  vcpkg-tool = callPackage ../by-name/vc/vcpkg-tool/package.nix {
2967:  doublecmd = callPackage ../by-name/do/doublecmd/package.nix {
6002:        ../by-name/me/meson/emulator-hook.sh
8995:  gtkclipblock = callPackage ../by-name/gt/gtkclipblock/package.nix { stdenv = gcc13Stdenv; };
<and many more>
```

There's a handful of them

Reply:
Oh but none of them are using libsForQt5.callPackage?

Reply:
> So just to see if I understood correctly... would it be better to migrate the packaging to the old way? and then make the call via libsForQt5.callPackage?

I think so.

Reply:
Ok, it ended up taking me a while, but it seems that calling the package in `pkgs/by-name/` directly from all-packages "works just fine". I even tested the application locally on my machine that pulls the `simulide` through an overlay and everything seems to "work as it should".

Reply:
The double quotes were due to the following error:
```bash
Run if result/bin/nixpkgs-check-by-name --base "$base" .; then
  if result/bin/nixpkgs-check-by-name --base "$base" .; then
    exit 0
  else
    exitCode=$?
    echo "To run locally: ./maintainers/scripts/check-by-name.sh $GITHUB_BASE_REF https://github.com/$GITHUB_REPOSITORY.git"
    exit "$exitCode"
  fi
  shell: /usr/bin/bash -e {0}
  env:
    mergedSha: a4e7e7803169340212267591c521158dfea0f190
    base: /tmp/tmp.5Ni4eCkKHd
    TMPDIR: /home/runner/work/_temp
pkgs.simulide: This attribute is manually defined (most likely in pkgs/top-level/all-packages.nix), which is only allowed if the definition is of the form `pkgs.callPackage pkgs/by-name/si/simulide/package.nix { ... }` with a non-empty second argument.
Validation failed, see above errors
To run locally: ./maintainers/scripts/check-by-name.sh master https://github.com/NixOS/nixpkgs.git
Error: Process completed with exit code 1.
```
Highlighting the following:
> This attribute is manually defined (most likely in pkgs/top-level/all-packages.nix), which is only allowed if the definition is of the form `pkgs.callPackage pkgs/by-name/si/simulide/package.nix { ... }` with a non-empty second argument.

 An error that is understandable, but that **causes me confusion** because wouldn't it be kind of the same as what I'm doing in practice? Because as far as I understand the use of `libsForQt5.callPackage` is precisely a way of passing arguments that `callPackage` by default would not pass.  Or am I mistaken?

Reply:
Anyway, I'm now moving the package to the old hierarchy, as this check is indeed failing. Just one moment...

Reply:
Okay, it passed the checks, and the package still seems to work as it should!